### PR TITLE
Typescript now reads type information without error. #968

### LINKED ISSUE #968: Error in types.d.ts TypeScript definitions
php-parser version 3.1.0-beta.11
tsc version 4.7.4

```
node_modules/php-parser/types.d.ts:327:9 - error TS2416: Property 'value' in type 'Encapsed' is not assignable to the same property in base type 'Literal'.
  Type 'EncapsedPart[]' is not assignable to type 'string | number | boolean | Node | null'.

327         value: EncapsedPart[];
            ~~~~~


Found 1 error in node_modules/php-parser/types.d.ts:327
```

Possible solution: extend Encapsed class directly from Expression and add `raw: string;` directly to Encapsed class.

### PR DESCRIPTION
**Pull request by ytetsuro**
This PR is fixed following.

- wrong jsdoc.
- wrong NullSafePropertyLookup constructor name.
- mergeable GitHub Actions to the wrong jsdoc.

fixed #968

### CODE REVIEW

**Comment by ytetsuro** [comment]
@czosel 

When you have time, I would be happy to review it for you.

**Comment by czosel** [comment]
@ytetsuro Thanks for contributing! Unfortunately I’m not very familiar with typescript and also swamped with work currently. Maybe @MaartenStaa or @cseufert could take a look?

File: src/ast/array.js [hunk 1]
```diff
@@ -31,7 +31,7 @@ const KIND = "array";
  *  ]
  * }
  * @extends {Expression}
- * @property {Entry|Expression|Variable} items List of array items
+ * @property {Entry[]|Expression[]|Variable[]} items List of array items
```
Comment (cseufert) [inline src/ast/array.js:34]: Should this actually be:
```
 * @property {Array<Entry|Expression|Variable>} items List of array items
```
As it I think the items can contain any of those.

**Review by cseufert — APPROVED** [review]
Looks great apart from query on items in array.

Thanks for all the clean-up work

**Comment by ytetsuro** [comment]
@MaartenStaa or @cseufert

I would be happy to rereview and merge when you have time.

**Review by cseufert — APPROVED** [review]
Looks good to me

**Comment by czosel** [comment]
Thanks again @ytetsuro and @cseufert!

**Comment by ytetsuro** [comment]
Thanks merged!

**Comment by czosel** [comment]
Released in `v3.1.0` :tada:

**Comment by ytetsuro** [comment]
Thanks released and releasing the notification!!

**Pull request merged**